chain/store/store.go

```diff
@@ -1444,8 +1428,8 @@ func (cs *ChainStore) Export(ctx context.Context, ts *types.TipSet, inclRecentRo
 		return xerrors.Errorf("failed to write car header: %s", err)
 	}
 
-	return cs.WalkSnapshot(ctx, ts, inclRecentRoots, skipOldMsgs, func(c cid.Cid) error {
-		blk, err := cs.bs.Get(c)
+	return cs.WalkSnapshot(ctx, ts, inclRecentRoots, skipOldMsgs, true, func(c cid.Cid) error {
+		blk, err := cs.chainBlockstore.Get(c)
```

Comment:
This can also be getting cids for the stateBlockstore, will this be an issue?

Reply:
Not right now, but we need some union abstraction for these cases.

Reply:
I implemented a union blockstore in 2047a74. This concern should be solved now.